tests/unit/io/GeoJSONReaderTest.cpp

```diff
@@ -0,0 +1,264 @@
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("MULTIPOLYGON (((40.000 40.000, 20.000 45.000, 45.000 30.000, 40.000 40.000)), ((20.000 35.000, 10.000 30.000, 10.000 10.000, 30.000 5.000, 45.000 20.000, 20.000 35.000), (30.000 20.000, 20.000 15.000, 20.000 25.000, 30.000 20.000)))", geom->toText());
+}
+
+template<>
+template<>
+void object::test<8>
+()
+{
+    std::string geojson { "{\"type\": \"GeometryCollection\",\"geometries\": [{\"type\": \"Point\",\"coordinates\": [40, 10]},{\"type\": \"LineString\",\"coordinates\": [[10, 10], [20, 20], [10, 40]]},{\"type\": \"Polygon\",\"coordinates\": [[[40, 40], [20, 45], [45, 30], [40, 40]]]}]}" };
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("GEOMETRYCOLLECTION (POINT (40.000 10.000), LINESTRING (10.000 10.000, 20.000 20.000, 10.000 40.000), POLYGON ((40.000 40.000, 20.000 45.000, 45.000 30.000, 40.000 40.000)))", geom->toText());
+}
+
+template<>
+template<>
+void object::test<9>
+()
+{
+    std::string geojson { "{\"type\":\"Feature\",\"geometry\":{\"type\":\"Point\",\"coordinates\":[-117.0,33.0]}}" };
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("POINT (-117.000 33.000)", geom->toText());
+
+}
+
+template<>
+template<>
+void object::test<10>
+()
+{
+    std::string geojson { "{\"type\":\"FeatureCollection\",\"features\":[{\"type\":\"Feature\",\"geometry\":{\"type\":\"Point\",\"coordinates\":[-117.0,33.0]}},{\"type\":\"Feature\",\"geometry\":{\"type\":\"Point\",\"coordinates\":[-122.0,45.0]}}]}" };
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("GEOMETRYCOLLECTION (POINT (-117.000 33.000), POINT (-122.000 45.000))", geom->toText());
+
+}
+
+template<>
+template<>
+void object::test<11>
+()
+{
+    std::string geojson { "{\"type\":\"Point\",\"coordinates\":[]}" };
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("POINT EMPTY", geom->toText());
+}
+
+template<>
+template<>
+void object::test<12>
+()
+{
+    std::string geojson { "{\"type\":\"LineString\",\"coordinates\":[]}" };
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("LINESTRING EMPTY", geom->toText());
+}
+
+template<>
+template<>
+void object::test<13>
+()
+{
+    std::string geojson { "{\"type\":\"Polygon\",\"coordinates\":[]}" };
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("POLYGON EMPTY", geom->toText());
+}
+
+template<>
+template<>
+void object::test<14>
+()
+{
+    std::string geojson { "{\"type\":\"MultiPoint\",\"coordinates\":[]}" };
+    GeomPtr geom(geojsonreader.read(geojson));
+    ensure_equals("MULTIPOINT EMPTY", geom->toText());
+}
+
+template<>
+template<>
+void object::test<15>
+()
+{
+    std::string geojson { "{\"type\":\"MultiLineString\",\"coordinates\":[]}" };
```

Comment:
Same as polygon, non-empty list of empty line strings? What happens with overly-wide dimensionality? 3-, 4-, 5- coordinates?

Reply:
I am adding unit test 28 for the first case and 29 for second.